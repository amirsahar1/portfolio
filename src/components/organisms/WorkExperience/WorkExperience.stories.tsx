/* eslint-disable import/no-anonymous-default-export */
import {
	WorkExperienceDesktop,
	WorkExperienceProps,
} from './WorkExperienceDesktop';

import { mockJobs } from '@mockdata';

export default {
	title: 'organisms/WorkExperience',
	component: WorkExperienceDesktop,
};

export const WorkExperienceStory = (args: WorkExperienceProps) => (
	<WorkExperienceDesktop {...args} />
);

WorkExperienceStory.storyName = 'WorkExperience';
WorkExperienceStory.args = {
	jobs: mockJobs,
};
