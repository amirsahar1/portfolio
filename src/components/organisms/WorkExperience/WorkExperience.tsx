'use client';

import { AnimatedTooltip } from '@components/atoms/AnimatedTooltip/animatedTooltip';
import { Container } from '@components/atoms/Container';
import { ContentBlock } from '@components/atoms/ContentBlock';
import { Timeline } from '@components/atoms/Timeline/timeline';
import { Education } from '@components/organisms/Education';

import LogoTs from '@root/public/assets/tools/Typescript.svg';
import LogoAws from '@root/public/assets/tools/aws.svg';
import LogoCypress from '@root/public/assets/tools/cypress.svg';
import LogoDocker from '@root/public/assets/tools/docker.svg';
import LogoMongoDb from '@root/public/assets/tools/mongodb.svg';
import LogoNest from '@root/public/assets/tools/nest.svg';
import LogoNext from '@root/public/assets/tools/nextjs.svg';
import LogoNode from '@root/public/assets/tools/node.svg';
import LogoNx from '@root/public/assets/tools/nx.svg';
import LogoReact from '@root/public/assets/tools/react-framework.svg';
import LogoStorybook from '@root/public/assets/tools/storybook.svg';
import { Education as EducationType } from '@root/src/types/education';
import type { Job as JobType } from '@types';
import dayjs from 'dayjs';
import React from 'react';
import 'react-responsive-carousel/lib/styles/carousel.min.css';

export interface WorkExperienceProps {
	jobs: JobType[];
	education: EducationType[];
}

const WorkExperience = ({ jobs, education }: WorkExperienceProps) => {
	const mappedJobs = jobs.map((job: JobType) => {
		const from = dayjs(job.fromDate);
		const to = dayjs(job.toDate);
		return {
			title: (
				<div className=" justify-items-center">
					<div className="mb-0">{`${from.format('YYYY')} - ${
						job.toDate ? to.format('YYYY') : 'Present'
					}`}</div>
					<div className="text-lg mb-2">{job.company.name}</div>
					<div>
						{job.clients && job.clients.length > 0 ? (
							<div className="flex flex-row md:flex-col gap-6">
								<AnimatedTooltip
									items={job.clients.map((cl, index) => {
										return {
											id: index,
											name: cl.name,
											image: cl.logo,
											url: cl.url,
										};
									})}
								/>
							</div>
						) : (
							<AnimatedTooltip
								items={[
									{
										id: 1,
										name: job.company.name,
										image: job.company.logo,
										url: job.company.url,
									},
								]}
							/>
						)}
					</div>
				</div>
			),
			content: (
				<div>
					<div className="text-xl md:text-2xl">{job.jobTitle}</div>
					<div className="text-neutral-800 dark:text-neutral-200 font-normal mb-8">
						<ContentBlock value={job.description} />
					</div>
				</div>
			),
		};
	});

	return (
		<>
			<h1 className="headline mt-8 text-3xl md:text-5xl lg:text-6xl md:leading-normal lg:leading-normal">
				Hey, I&apos;m Amir Sahar
			</h1>
			<h2 className="mt-2 text-xl font-bold md:text-2xl">
				Senior Software Engineer from Israel
			</h2>
			<p className="mt-8">
				I am 28 years old and have been a web developer for as long as I can
				think. The technologies I work with are JavaScript, HTML and CSS with a
				focus on the frameworks React.js, Next.js, Node and Express. When it
				comes to software development, I believe in the principles of
				efficiency, scalability and speed without compromising quality.
			</p>
			<section id="experience" className="mt-24">
				<h1 className="headline text-3xl md:text-5xl lg:text-6xl xs:mb-0 md:-mb-16 ">
					Experience
				</h1>
				<div className="w-full">
					<Timeline data={mappedJobs} />
				</div>
			</section>
			<h2 className="headline mb-12 mt-24 text-center text-5xl">Education</h2>
			<Education education={education} />
			<h2 className="headline mb-12 mt-24 text-center text-5xl">Tools</h2>
			<section id="tools" className="mb-24">
				<Container>
					<div className="mx-auto mt-8 flex max-w-5xl flex-wrap items-center justify-center gap-x-32 gap-y-4">
						<LogoNext className="w-32" aria-label="Next.js" />
						<LogoMongoDb className="w-32 h-32" aria-label="Mongo" />
						<LogoNx className="w-16" aria-label="Sanity" />
						<LogoNode className="w-32" aria-label="Node.js" />
						<LogoNest className="w-40" aria-label="Nest.js" />
						<LogoAws className="w-32 h-32" aria-label="Aws" />
						<LogoStorybook className="w-32" aria-label="Storybook" />
						<LogoReact className="w-20" aria-label="React" />
						<LogoCypress className="w-32" aria-label="Cypress" />
						<LogoTs className="w-20" aria-label="TypeScript" />
						<LogoDocker className="w-24" aria-label="Docker" />
					</div>
				</Container>
			</section>
		</>
	);
};

export { WorkExperience };
