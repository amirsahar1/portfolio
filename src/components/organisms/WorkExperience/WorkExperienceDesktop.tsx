'use client';

import { Button } from '@components/atoms/Button';
import { ContentBlock } from '@components/atoms/ContentBlock';
import { Job } from '@components/molecules/Job';

import type { Job as JobType } from '@types';
import React, { useEffect, useState } from 'react';
import { AiOutlineArrowLeft, AiOutlineArrowRight } from 'react-icons/ai';
import { Carousel } from 'react-responsive-carousel';
import 'react-responsive-carousel/lib/styles/carousel.min.css';

export interface WorkExperienceProps {
	jobs: JobType[];
}

const WorkExperienceDesktop = ({ jobs }: WorkExperienceProps) => {
	const Spacer = () => <div className="md:w-1/8 w-12 shrink-0 grow-0" />;
	return (
		<div className="px-2 lg:w-2/3 mx-auto">
			<Carousel
				showThumbs={false}
				interval={4000}
				swipeable={false}
				axis="horizontal"
				className="relative"
				renderArrowPrev={(clickHandler, hasPrev, labelPrev) =>
					hasPrev && (
						<div className="absolute flex justify-center items-center top-0 bottom-0  p-3 cursor-pointer z-20">
							<Button
								onClick={clickHandler}
								className="hover:shadow-none opacity-60 active:opacity-100"
							>
								<AiOutlineArrowLeft />
							</Button>
						</div>
					)
				}
				renderArrowNext={(clickHandler, hasNext, labelPrev) =>
					hasNext && (
						<div className="absolute flex justify-center items-center top-0 bottom-0  p-3 cursor-pointer z-20 right-0">
							<Button
								onClick={clickHandler}
								className="shadow-none opacity-60 active:opacity-100"
							>
								<AiOutlineArrowRight />
							</Button>
						</div>
					)
				}
				showArrows={true}
			>
				{jobs.map(
					({ company, jobTitle, fromDate, toDate, skills, description }) => (
						<div
							className="w-[calc(100% + 2rem)] mx-4 text-start"
							key={`${jobTitle}-${company.name}`}
						>
							<Job
								company={company}
								jobTitle={jobTitle}
								fromDate={fromDate}
								toDate={toDate}
								skills={skills}
							>
								<ContentBlock value={description} />
							</Job>
						</div>
					)
				)}
			</Carousel>
			<Spacer />
		</div>
	);
};

export { WorkExperienceDesktop };
