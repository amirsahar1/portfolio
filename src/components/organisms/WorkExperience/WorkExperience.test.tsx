import { WorkExperienceDesktop } from '../WorkExperience';

import { mockJobs } from '@mockdata';
import { render } from '@testing-library/react';

jest.mock('@components/atoms/ContentBlock', () => ({
	ContentBlock: () => <div id="contentBlock" />,
}));

describe('WorkExperience', () => {
	it('renders correctly', () => {
		const { container } = render(
			<WorkExperienceDesktop jobs={[mockJobs[0]]} />
		);
		expect(container).toMatchSnapshot();
	});
});
