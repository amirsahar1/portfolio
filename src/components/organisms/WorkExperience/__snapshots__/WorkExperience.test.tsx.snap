// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkExperience renders correctly 1`] = `
<div>
  <div
    class="px-2 lg:w-2/3 mx-auto"
  >
    <div
      class="carousel-root relative"
    >
      <div
        class="carousel carousel-slider"
        style="width: 100%;"
      >
        <ul
          class="control-dots"
        >
          <li
            aria-label="slide item 1"
            class="dot selected"
            role="button"
            tabindex="0"
            value="0"
          />
        </ul>
        <div
          class="slider-wrapper axis-horizontal"
        >
          <ul
            class="slider animated"
            style="transform: translate3d(0,0,0); transition-duration: 350ms;"
          >
            <li
              class="slide selected previous"
            >
              <div
                class="w-[calc(100% + 2rem)] mx-4 text-start"
              >
                <div
                  class="mt-12 w-full glass-light dark:glass-dark mb-6 rounded-xl px-8 py-6 drop-shadow-lg"
                >
                  <div
                    class="-mt-12 mb-4 flex w-full justify-center"
                  >
                    <div
                      class="w-20"
                    >
                      <img
                        alt="Holland & Barrett"
                        class="rounded-md"
                        data-nimg="1"
                        decoding="async"
                        height="64"
                        loading="lazy"
                        src="/_next/image?url=https%3A%2F%2Fjacobherper.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252F220cf610ee8a9bc6f821a5a3285ca7c224536cb8-128x128.jpg%26w%3D128%26q%3D75&w=128&q=75"
                        srcset="/_next/image?url=https%3A%2F%2Fjacobherper.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252F220cf610ee8a9bc6f821a5a3285ca7c224536cb8-128x128.jpg%26w%3D128%26q%3D75&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fjacobherper.com%2F_next%2Fimage%3Furl%3Dhttps%253A%252F%252Fcdn.sanity.io%252Fimages%252Fyq5nnpfd%252Fproduction%252F220cf610ee8a9bc6f821a5a3285ca7c224536cb8-128x128.jpg%26w%3D128%26q%3D75&w=128&q=75 2x"
                        style="color: transparent;"
                        width="64"
                      />
                    </div>
                  </div>
                  <div
                    class="flex items-start gap-4 md:items-center"
                  >
                    <div
                      class="mb-2 w-full"
                    >
                      <h3
                        class="text-center font-bold"
                      >
                        Senior Software Engineer
                      </h3>
                      <h4
                        class="text-center"
                      >
                        <a
                          class="underlined relative"
                          href="https://www.hollandandbarrett.com/"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          Holland & Barrett
                        </a>
                      </h4>
                      <p
                        class="mb-4 text-center text-sm text-slate-500"
                      >
                        Aug 2020 â€” present
                      </p>
                      <p
                        class="hidden text-sm md:block md:text-base"
                      >
                        <strong>
                          Technologies: 
                        </strong>
                        TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
                      </p>
                    </div>
                  </div>
                  <p
                    class="text-sm md:hidden"
                  >
                    <strong>
                      Technologies: 
                    </strong>
                    TypeScript, React, Next.js, Redux, GraphQL, REST API, Jest, Cypress
                  </p>
                  <div
                    id="contentBlock"
                  />
                </div>
              </div>
            </li>
          </ul>
        </div>
        <p
          class="carousel-status"
        >
          1 of 1
        </p>
      </div>
    </div>
    <div
      class="md:w-1/8 w-12 shrink-0 grow-0"
    />
  </div>
</div>
`;
